### PR TITLE
DSPDC-68 Init server for data explorer API

### PR DESCRIPTION
**Pull request by danxmoran**
Result is running in the back-end here: http://broad-gdr-encode.appspot.com/. Be warned it's *painfully* slow, but I wanted to get it merged in.

### CODE REVIEW

File: encode/explorer/src/main/scala/org/broadinstitute/gdr/encode/explorer/db/DbClient.scala [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+           |from (
+           |  select unnest($column) as v
+           |  from ${table.entryName}
+           |  where $column is not null
+           |) as v
+           |group by v""".stripMargin
+      )
+      .query[(String, Long)]
+      .to[List]
+      .transact(transactor)
```
Comment (ahaessly) [inline encode/explorer/src/main/scala/org/broadinstitute/gdr/encode/explorer/db/DbClient.scala]: should you validate that the column is an array column? or is this checked somewhere else?

Comment (danxmoran) [reply encode/explorer/src/main/scala/org/broadinstitute/gdr/encode/explorer/db/DbClient.scala]: This is now validated on startup, see [here](https://github.com/broadinstitute/gdr-ingest/pull/28/files#diff-e584654d81fe268290fbc94f5a05c475R61) and [here](https://github.com/broadinstitute/gdr-ingest/pull/28/files#diff-bed0dcb4525c584b45e18157f38e7da5R54).

File: encode/explorer/src/main/scala/org/broadinstitute/gdr/encode/explorer/facets/FacetsController.scala [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        val col = f.column
+        if (dbFields.contains(col)) {
+          ().pure[M]
+        } else {
+          val err: Throwable = new IllegalStateException(
+            s"No such field in table '${table.entryName}': $col"
+          )
+          err.raiseError[M, Unit]
+        }
+      }
```
Comment (ahaessly) [inline encode/explorer/src/main/scala/org/broadinstitute/gdr/encode/explorer/facets/FacetsController.scala]: probably not important now, but if we are expecting to have a large number of field to validate, you might want to try to validate all fields and then return errors for all the ones that do not pass validation.

Comment (danxmoran) [reply encode/explorer/src/main/scala/org/broadinstitute/gdr/encode/explorer/facets/FacetsController.scala]: Agreed. I've got this half-refactored on a WIP branch, will try to make it work there.

**Pull request merged**